AsyncDisplayKit/ASVideoNode.mm

```diff
@@ -505,7 +508,7 @@ - (AVAsset *)asset
 
 - (void)_setAndFetchAsset:(AVAsset *)asset url:(NSURL *)assetURL
 {
-  [self clearFetchedData];
+  [self _clearPlayer]; // Clear the player but not the underlying ASNetworkImageNode to avoid clearing the placeholder image and showing the background before the video starts.
   _asset = asset;
```

Comment:
Ahh I didn't realized that a placeholder image was generated if ASVideoNode.URL was nil. That makes sense to check if URL is set:
```objective-c
- (void)_setAndFetchAsset:(AVAsset *)asset url:(NSURL *)assetURL
{
  if (self.URL != nil) {
    [self _clearPlayer]; // Clear the player but not the underlying ASNetworkImageNode to avoid clearing the placeholder image and showing the background before the video starts.
  } else {
    [self clearFetchedData]; // Clear player and underlying ASNetworkImageNode with generated placeholder
  }
  _asset = asset;
  _assetURL = assetURL;
  [self setNeedsDataFetch];
}
```


If we did it this way the tests should pass without modification.